### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by leetdev**
I have made a few more improvements to the code:

1. Made sure all translatable strings in the code are evaluated the same way PHP evaluates them when running the code. That includes all sorts of escape sequences, which may or may not differ based on whether you're using single or double quotes.   _Edit:_ Sadly, this caused Codacy to complain over the use of the `eval` function, but doing this in any other way would likely require a serious overhaul of the way the strings are parsed.   _Edit 2:_ Avoided using `eval` by using `stripcslashes` for double quoted strings, and simply replacing escaped single quotes and backslashes otherwise, which should mimic the behavior of PHP exactly.
2. Previously, when a string included a closing parenthesis, the parser would count it as the ending parenthesis of the translation function, which would break the detection of the string (usually resulting in a truncated result, or even no detection at all in certain cases). I refactored the parsing function to account for such cases properly.
3. Added `JSON_UNESCAPED_UNICODE` flag to the `json_encode()` function call when writing the translations file. This avoids ugly escape sequences in the results as well as the case where existing keys were not found due to previously escaped unicode characters being present in the JSON file but unescaped in the actual string used in the translation function call.
4. A minor one, but spaces before and after the opening parentheses of the translation function calls are also now supported.

### CODE REVIEW

File: src/Translator/TranslationScanner.php [hunk 1]
```diff
@@ -54,14 +54,12 @@ private function getFileContent(string $filePath): string
...
-        preg_match_all("#{$functionName}\((.*?)\)#", $content, $matches);
+        preg_match_all("#{$functionName} *\( *((['\"])((?:\\\\\\2|.)*?)\\2)#", $content, $matches);
 
         $matches = $matches[1] ?? [];
 
         return array_reduce($matches, function (array $keys, string $match) {
-            preg_match("#(['\"])((?:\\\\\\1|.)*?)\\1#", $match, $matchKey);
-
-            $key = $matchKey[2] ?? '';
+            eval("\$key = $match;");
```
Comment (thiagocordeiro) [inline src/Translator/TranslationScanner.php]: `eval` is not safe and could potentially make this to be known as an unsafe library, please check if `$key = stripcslashes(trim($match, '\'\"'));` solves this issue

Comment (leetdev) [reply src/Translator/TranslationScanner.php]: The problem I'm having currently is that I have a string like this (slightly modified from Laravel's default email template): `"If you’re having trouble clicking the \":actionText\" button, copy and paste the URL below\ninto your web browser: [:actionURL](:actionURL)"`. Now, if we scan it and write out as JSON, the backslashes and double quotes will be escaped, so we end up with `"If you’re having trouble clicking the \\\":actionText\\\" button, copy and paste the URL below\\ninto your web browser: [:actionURL](:actionURL)": ""`, which is not what we want. The string should stay the same, since it's a double quoted string originally. But if we had something similar in single quotes, the behavior would be different - then we would actually need the backslashes escaped.

`stripcslashes` certainly doesn't help in this situation. I think we need to have separate logic for single and double quoted strings. The logic needs to match PHP's evaluation of the strings exactly, as only then will the keys match the actual strings passed to the translation function. That's why I went with `eval` as this is the simplest to use, but I am aware using it is considered bad practice for obvious reasons.

In conclusion - this needs some additional thought. I will try to figure something out tomorrow. If you have any ideas, feel free to share.

Comment (leetdev) [reply src/Translator/TranslationScanner.php]: Avoided using `eval` by using `stripcslashes` for double quoted strings, and simply replacing escaped single quotes and backslashes otherwise, which should mimic the behavior of PHP exactly.

**Comment by thiagocordeiro** [comment]
`eval` was a clever solution, but it's better like this once we don't use an unsafe function, thanks @leetdev for this amazing improvements

**Comment by thiagocordeiro** [comment]
Changes available on [v1.0.3](https://github.com/thiagocordeiro/laravel-translator/tree/v1.0.3)
